src/com.target.lorri.internal.varlink

```diff
@@ -0,0 +1,72 @@
+# The interface `lorri daemon` exposes.
+interface com.target.lorri.internal
+
+# WatchShell instructs the daemon to evaluate a Nix expression and re-evaluate
+# it when it or its dependencies change.
+method WatchShell(shell_nix: ShellNix) -> ()
+
+# ShellNix describes the Nix expression which evaluates to a development
+# environment.
+type ShellNix (
+  # The absolute path of a Nix file specifying the project environment.
+  path: string
+)
+
+type Reason (
+    kind: (project_added, ping_received, files_changed, unknown),
+    project: ?ShellNix, # only present if kind == project_added
+    files: ?[]string,   # only present if kind == files_changed
+    debug: ?string      # only present if kind == unknown
+)
+
+type Outcome (
+    project_root: string
+)
+
+type Failure (
+    kind: (io, spawn, exit, output),
+    msg: ?string,   # only present if kind in (io, spawn)
+    cmd: ?string,   # only present if kind in (spawn, exit)
+    status: ?int,   # only present if kind == exit
+    logs: ?[]string # only present if kind == exit
+)
+
+# WatchServices establishes a stream with the daemon. Initially, the daemon
+# evaluates the given services definition to an array of Command objects and
+# sends a reply for each of them. After this initial evaluation, the daemon
+# watches the services definition and its dependencies for changes,
+# re-evaluates it as appropriate and sends a reply for each Command again.
+#
+# This is a streaming RPC. The daemon only accepts client calls with the "more"
+# property set - see https://varlink.org/Method-Call.
+# TODO: Implement WatchServices
+#method WatchServices(services_nix: ServicesNix) -> (service: Service)
+
+# ServicesNix describes the Nix expression which evaluates to a list of
+# services.
+type ServicesNix (
+  # The absolute path of a Nix file specifying the services to be run. This Nix
+  # file must evaluate to a JSON document of type []Command, that is, an array
+  # of objects whose properties are described by the Command type.
+  path: string
+)
+
+# Service describes an individual service to be run.
+type Service (
+  # The user-friendly name of the service. This is used for identification
+  # purposes too: only a single instance of a service with a particular name is
+  # run at any one time.
+  name: string,
+
+  # How to run the service.
+  command: Command
+)
+
+# Command describes how to run a terminal application.
+type Command (
+  # The path of the command binary.
+  program: string,
+
+  # Arguments to be passed to the binary.
+  args: []string
+)
```

Comment:
```suggestion
```

Services aren't a thing anymore - there is no need to copy this over. See https://github.com/target/lorri/issues/404 and https://github.com/target/lorri/pull/405.

Reply:
Skeptic though I was, I find that kind of a shame. I wonder if a separate tool might be worth examining? Or if something based on runit/s6 might be added to the pipeline?

Reply:
If you have ideas, feel free to join the discussion at https://github.com/target/lorri/issues/104 and https://github.com/target/lorri/issues/108 :)

However, AFAICT nobody is working on this currently.